aio_pika/tools.py

```diff
@@ -2,6 +2,38 @@
 from functools import partial
 
 
+def iscoroutinepartial(fn):
```

Comment:
Hmm.. You are right it's so slow.

```python
In [3]: def foo():
   ...:     pass
   ...:

In [4]: %timeit asyncio.coroutine(foo)
The slowest run took 4.28 times longer than the fastest. This could mean that an intermediate result is being cached.
100000 loops, best of 3: 8.04 µs per loop

In [5]: def iscoroutinepartial(fn):
   ...:     while True:
   ...:         parent = fn
   ...:         fn = getattr(parent, 'func', None)
   ...:         if fn is None:
   ...:             break
   ...:     return asyncio.iscoroutinefunction(parent)
   ...:

In [6]: %timeit iscoroutinepartial(foo)
1000000 loops, best of 3: 1.59 µs per loop